pkg/utils/dataset/lifecycle/node.go

```diff
@@ -182,5 +179,52 @@ func LabelCacheNode(nodeToLabel v1.Node, runtimeInfo base.RuntimeInfoInterface,
 		return err
 	}
 
+	// Wait at most 30s for cache in controller-runtime successfully catching up with api-server
+	// This is to ensure the controller can get up-to-date cluster status during the following scheduling
+	// loop.
+	if err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
+		node := &v1.Node{}
+		namespacedName := types.NamespacedName{
+			Name: nodeName,
+		}
+		if err := client.Get(context.TODO(), namespacedName, node); err != nil {
+			return false, fmt.Errorf("failed to get node: %w", err)
+		}
+		return reflect.DeepEqual(toUpdate, node), nil
```

Comment:
Should deepEqual? Or just check the target label?

Reply:
You're right. `deepEqual` is not appropriate here. Replaced with a label existence check.